src/MagnumPlugins/BasisImporter/BasisImporter.cpp

```diff
@@ -0,0 +1,217 @@
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNETCION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "BasisImporter.h"
+
+#include <cstring>
+#include <algorithm>
+#include <basisu_transcoder.h>
+
+#include <Corrade/Utility/ConfigurationValue.h>
+#include <Corrade/Utility/String.h>
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/Utility/Debug.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Trade/ImageData.h>
+
+namespace Magnum { namespace Trade {
+
+namespace {
+
+/* Map BasisTranscodingType to Magnum::CompressedPixelFormat */
+CompressedPixelFormat getTextureFormat(BasisTranscodingType type, bool hasAlpha) {
+    switch(type) {
+        case BasisTranscodingType::Etc1:
+        case BasisTranscodingType::Etc2:
+            return (hasAlpha) ? CompressedPixelFormat::Etc2RGBA8Unorm
+                : CompressedPixelFormat::Etc2RGB8Unorm;
+        case BasisTranscodingType::Bc1:
+            return (hasAlpha) ? CompressedPixelFormat::Bc1RGBAUnorm
+                : CompressedPixelFormat::Bc1RGBUnorm;
+        case BasisTranscodingType::Bc3:
+            CORRADE_ASSERT(hasAlpha, "Bc3 always has alpha, but non-alpha was requested",
+                CompressedPixelFormat::Bc3RGBAUnorm);
+            return CompressedPixelFormat::Bc3RGBAUnorm;
+        case BasisTranscodingType::Bc4:
+            return CompressedPixelFormat::Bc4RUnorm;
+        case BasisTranscodingType::Bc5:
+            return CompressedPixelFormat::Bc5RGUnorm;
+        case BasisTranscodingType::Bc7M6OpaqueOnly:
+            CORRADE_ASSERT(hasAlpha, "Bc7 is opaque only, but alpha was requested",
+                CompressedPixelFormat::Bc7RGBAUnorm);
+            return CompressedPixelFormat::Bc7RGBAUnorm;
+        case BasisTranscodingType::Pvrtc1_4OpaqueOnly:
+            CORRADE_ASSERT(hasAlpha, "Pvrtc1 is opaque only, but alpha was requested",
+                CompressedPixelFormat::PvrtcRGB4bppUnorm);
+            return CompressedPixelFormat::PvrtcRGB4bppUnorm;
+        default:
+            CORRADE_ASSERT_UNREACHABLE();
+    }
+}
+
+basist::transcoder_texture_format getTranscoderTextureFormat(BasisTranscodingType type) {
+    constexpr basist::transcoder_texture_format formats[] = {
+        basist::transcoder_texture_format::cTFETC1,
+        basist::transcoder_texture_format::cTFETC2,
+        basist::transcoder_texture_format::cTFBC1,
+        basist::transcoder_texture_format::cTFBC3,
+        basist::transcoder_texture_format::cTFBC4,
+        basist::transcoder_texture_format::cTFBC5,
+        basist::transcoder_texture_format::cTFBC7_M6_OPAQUE_ONLY,
+        basist::transcoder_texture_format::cTFPVRTC1_4_OPAQUE_ONLY
+    };
+
+    return formats[Int(type)];
+}
```

Comment:
What about doing just `TranscoderTextureFormat[Int(type)]` (and having the array file-local) instead of calling the function?